server/db/schema/hostels.schema.ts

```diff
@@ -0,0 +1,45 @@
+import { relations } from 'drizzle-orm';
+import { pgTable, serial, text, timestamp, varchar } from 'drizzle-orm/pg-core';
+
+import { staff } from '.';
+
+export const hostels = pgTable('hostels', {
+  id: serial('id').primaryKey(),
+  address: text('address').notNull(),
+  name: varchar('name', { length: 256 }).notNull(),
+  email: varchar('email', { length: 256 }).notNull(),
+  telephone: varchar('telephone', { length: 13 }).notNull(),
+  alternateTelephone: varchar('alternate_telephone', { length: 13 }),
+  type: varchar('type', {
+    enum: ['boys', 'girls'],
+  }).notNull(),
+  createdAt: timestamp('created_at').defaultNow().notNull(),
+  updatedAt: timestamp('updated_at')
+    .$onUpdate(() => new Date())
+    .notNull(),
+});
+
+export const hostelRelations = relations(hostels, ({ many }) => ({
+  posts: many(hostelPosts),
+}));
+
+export const hostelPosts = pgTable('hostel_posts', {
+  hostelId: serial('hostel_id')
+    .references(() => hostels.id)
+    .notNull(),
+  staffId: serial('staff_id')
+    .references(() => staff.id)
+    .notNull(),
+  post: varchar('post', { enum: ['warden', 'deputy-warden'] }).notNull(),
```

Comment:
IDK about deputy wardens, but wardens are not staff, they're faculty.

Reply:
so should i make a single relation for staff and faculty or two different ones, cause practically same columns need to be retrieved in either

Reply:
Given that only the warden/deputy-warden is a faculty member, I think it's best to offload that post to 2 columns in the `hostels` table (`wardenId`, and `deputyWardenId` I guess).

Then, leave this table as is, updating the enum to mention other values.